atproto/identity/cache_catalog.go

```diff
@@ -0,0 +1,209 @@
+package identity
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+type CacheCatalog struct {
+	HitTTL        time.Duration
+	ErrTTL        time.Duration
+	Inner         Catalog
+	mutex         sync.RWMutex
+	handleCache   map[syntax.Handle]HandleEntry
+	identityCache map[syntax.DID]IdentityEntry
+}
+
+type HandleEntry struct {
+	Updated time.Time
+	DID     syntax.DID
+	Err     error
+}
+
+type IdentityEntry struct {
+	Updated  time.Time
+	Identity *Identity
+	Err      error
+}
+
+var _ Catalog = (*CacheCatalog)(nil)
+
+func NewCacheCatalog(inner Catalog) CacheCatalog {
+	// TODO: these are kind of arbitrary default values...
+	hitTTL, err := time.ParseDuration("1h")
+	if err != nil {
+		panic(err)
+	}
+	errTTL, err := time.ParseDuration("2m")
+	if err != nil {
+		panic(err)
+	}
```

Comment:
It's safe with constant strings like this but still gives me pause to see a `panic` in a library. Maybe just change it to be`func NewCacheCatalog(inner Catalog, hitTTL, errTTL time.Duration) CacheCatalog` or whatever from the start?

Reply:
Cross reference about pain of doing static error handling: https://github.com/golang/go/issues/54297

Replaced with static values in nanoseconds, which I verified were equal to the parsed values.

Reply:
Why not just do `time.Hour * 1` and `time.Minute * 2`?

Reply:
even better!